src/ui/public/dialogues/__tests__/safe_confirm.js

```diff
@@ -45,35 +47,45 @@ describe('ui/safe_confirm', function () {
   });
 
   context('after timeout completes', function () {
-    it('$window.confirm is invoked with message', function () {
+    it('confirmation dialog is loaded to dom with message', function () {
       $timeout.flush();
-      expect($window.confirm.calledWith(message)).to.be(true);
+      const confirmationDialogElement = document.querySelector('.confirm-dialogue');
+      expect(!!confirmationDialogElement).to.be(true);
+      const htmlString = confirmationDialogElement.innerHTML;
+
+      expect(htmlString.indexOf(message)).to.be.greaterThan(0);
     });
 
     context('when confirmed', function () {
       it('promise is fulfilled with true', function () {
         $timeout.flush();
 
         let value;
-        promise.then(function (v) {
+        promise.then((v) => {
           value = v;
         });
-        $rootScope.$apply();
+
+        const confirmationDialogElement = document.querySelector('.confirm-dialogue');
+        const okayButton = confirmationDialogElement.getElementsByTagName('button')[0];
```

Comment:
We should also use data-test-subj selectors to find these buttons. This will uncouple our tests from our elements and from the order in which they appear.